### PR TITLE
Panel Web UI Documentation

### PR DESCRIPTION
**Pull request by MarcSkovMadsen**
## This PR Adds documentation for adding a Panel web ui

Adds documentation for adding a web ui using the [Panel](https://panel.holoviz.org) framework which is part of the features requested in https://github.com/Lightning-AI/lightning/issues/13335.

This adds a basic example only. I intend to add an intermediate example and links to some published examples later. But for now I would like to keep it simple as this is my first contribution to the project.

I am not able to find instructions for installing the requirements and building the docs on my constrained windows machine. So I've only inspected the .rst files manually using VS Code extensions. I've tried to inspect the artifacts build but could not find the dash, panel and streamlit documentation in there.

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by MarcSkovMadsen** [comment]
The app is deployed here https://01g6z4w0fgxzp3npc9cr2dmthh.litng-ai-03.litng.ai/view/Home

panel-lightning-basic

File: docs/source-app/workflows/add_web_ui/panel/basic.rst [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+.. figure:: https://cdn.jsdelivr.net/gh/MarcSkovMadsen/panel-lightning-basic@main/assets/panel-lightning-basic.gif
+   :alt: Basic Panel Lightning App
+
+   Basic Panel Lightning App
+
+First **create a file named app.py** with the app content:
+
+.. code:: bash
+
+    import lightning_app as lapp
```
Comment (lantiga) [inline docs/source-app/workflows/add_web_ui/panel/basic.rst:47]: We generally use
```
import lightning as L
```
and then refer to `L.LightningWork`, etc

File: docs/source-app/workflows/add_web_ui/panel/basic.rst [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+        length = pn.widgets.IntSlider(value=5, start=1, end=10, name="Length")
+        plot = pn.bind(get_plot, length=length)
+        component = pn.Column(length, plot)
+        template = pn.template.FastListTemplate(
+            title="⚡ Hello Panel + Lightning ⚡", main=[component], accent=ACCENT
+        )
+        return template
+
+
+    class LitPanel(lapp.LightningWork):
```
Comment (lantiga) [inline docs/source-app/workflows/add_web_ui/panel/basic.rst:92]: `L.LightningWork` (see previous comment)

File: docs/source-app/workflows/add_web_ui/panel/basic.rst [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+            pn.serve(
+                {"/": self._fast_initial_view},
+                port=self.port,
+                address=self.host,
+                websocket_origin="*",
+                show=False,
+            )
+
+
+    class LitApp(lapp.LightningFlow):
```
Comment (lantiga) [inline docs/source-app/workflows/add_web_ui/panel/basic.rst:115]: `L.LightningFlow`

File: docs/source-app/workflows/add_web_ui/panel/basic.rst [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+
+        def run(self):
+            self.lit_panel.run()
+
+        def configure_layout(self):
+            tab1 = {"name": "Home", "content": self.lit_panel}
+            return tab1
+
+
+    app = lapp.LightningApp(LitApp())
```
Comment (lantiga) [inline docs/source-app/workflows/add_web_ui/panel/basic.rst:128]: `L.LightningApp`

File: docs/source-app/workflows/add_web_ui/panel/basic.rst [hunk 5]
```diff
@@ -0,0 +1,329 @@
...
+2. Add the Panel app to a lightning component
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Add the Panel app to the run method of a ``LightningWork`` component and run the server on that
+component's **host** and **port**.
+
+.. code:: python
+    :emphasize-lines: 7, 8, 10
+
+    class LitPanel(lapp.LightningWork):
```
Comment (lantiga) [inline docs/source-app/workflows/add_web_ui/panel/basic.rst:259]: `L.LightningWork`

File: docs/source-app/workflows/add_web_ui/panel/basic.rst [hunk 6]
```diff
@@ -0,0 +1,329 @@
...
+                {"/": self._fast_initial_view},
+                port=self.port,
+                address=self.host,
+                websocket_origin="*",
+                show=False,
+            )
+
+Please note we need to wrap the `view` function by the `_fast_initial_view` such that the first,
+initial request done by the lightning server is fast. Otherwise it will continue requesting a
+response and never load the page.
```
Comment (lantiga) [inline docs/source-app/workflows/add_web_ui/panel/basic.rst:283]: I would mention this behavior will likely change in the future and doing this won't be necessary.

File: docs/source-app/workflows/add_web_ui/panel/basic.rst [hunk 7]
```diff
@@ -0,0 +1,329 @@
...
+3. Route the UI in the root component
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The final step, is to tell the Root component in which tab to render this component's UI.
+In this case, we render the ``LitPanel`` UI in the ``Home`` tab of the application.
+
+.. code:: python
+    :emphasize-lines: 4, 7, 10
+
+        class LitApp(lapp.LightningFlow):
```
Comment (lantiga) [inline docs/source-app/workflows/add_web_ui/panel/basic.rst:296]: `L.LightningFlow`

File: docs/source-app/workflows/add_web_ui/panel/basic.rst [hunk 8]
```diff
@@ -0,0 +1,329 @@
...
+
+            def run(self):
+                self.lit_panel.run()
+
+            def configure_layout(self):
+                tab1 = {"name": "Home", "content": self.lit_panel}
+                return tab1
+
+
+        app = lapp.LightningApp(LitApp())
```
Comment (lantiga) [inline docs/source-app/workflows/add_web_ui/panel/basic.rst:309]: `L.LightningApp`

**Review by lantiga — COMMENTED** [review]
Looks great @MarcSkovMadsen!
Just so you know, most people at Lightning will be out this coming week for a well deserved post-launch break, so this will likely take a few days to get merged.

File: docs/source-app/workflows/add_web_ui/panel/basic.rst (hunk 1 above)
Comment (krshrimali) [inline docs/source-app/workflows/add_web_ui/panel/basic.rst:47]:
```suggestion
    import lightning as L
```

File: docs/source-app/workflows/add_web_ui/panel/basic.rst (hunk 2 above)
Comment (krshrimali) [inline docs/source-app/workflows/add_web_ui/panel/basic.rst:92]:
```suggestion
    class LitPanel(L.LightningWork):
```

File: docs/source-app/workflows/add_web_ui/panel/basic.rst (hunk 3 above)
Comment (krshrimali) [inline docs/source-app/workflows/add_web_ui/panel/basic.rst:115]:
```suggestion
    class LitApp(L.LightningFlow):
```

File: docs/source-app/workflows/add_web_ui/panel/basic.rst (hunk 5 above)
Comment (krshrimali) [inline docs/source-app/workflows/add_web_ui/panel/basic.rst:259]:
```suggestion
    class LitPanel(L.LightningWork):
```

File: docs/source-app/workflows/add_web_ui/panel/basic.rst (hunk 7 above)
Comment (krshrimali) [inline docs/source-app/workflows/add_web_ui/panel/basic.rst:296]:
```suggestion
        class LitApp(L.LightningFlow):
```

File: docs/source-app/workflows/add_web_ui/panel/basic.rst (hunk 8 above)
Comment (krshrimali) [inline docs/source-app/workflows/add_web_ui/panel/basic.rst:309]:
```suggestion
        app = L.LightningApp(LitApp())
```

**Review by krshrimali — COMMENTED** [review]
Following @lantiga's nit's on imports - added those as suggestions, thanks for your work on this! :tada:

**Comment by MarcSkovMadsen** [comment]
This PR will be super seeded by https://github.com/Lightning-AI/lightning/pull/13531, where I contribute a PanelFrontend and make basic and intermediate documentation.

I will apply the feedback on this PR to the other PR.

**Pull request closed without merging**